pkg/schema/definitions/handler.go

```diff
@@ -130,13 +147,52 @@ func (s *SchemaDefinitionHandler) indexSchemaNames(models proto.Models, groups *
 			// we can safely continue
 			continue
 		}
+		schemaID := converter.GVKToSchemaID(*gvk)
 		prefVersion := preferredResourceVersions[gvk.Group]
-		// if we don't have a known preferred version for this group or we are the preferred version
-		// add this as the model name for the schema
+		// we always add the preferred version to the map. However, if this isn't the preferred version the preferred group could
+		// be missing this resource (e.x. v1alpha1 has a resource, it's removed in v1). In those cases, we add the model name
+		// only if we don't already have an entry. This way we always choose the preferred, if possible, but still have 1 version
+		// for everything
 		if prefVersion == "" || prefVersion == gvk.Version {
-			schemaID := converter.GVKToSchemaID(*gvk)
 			schemaToModel[schemaID] = modelName
+		} else {
+			_, ok := schemaToModel[schemaID]
+			if !ok {
+				schemaToModel[schemaID] = modelName
+			}
 		}
 	}
 	return schemaToModel
 }
+
+// baseSchemaToDefinition converts a given schema to the definition map. This should only be used with baseSchemas, whose definitions
+// are expected to be set by another application and may not be k8s resources.
+func baseSchemaToDefinition(schema types.APISchema) map[string]definition {
```

Comment:
Is my understanding correct that this doesn't need to be recursive because a field has either a primitive type or:
- array whose subtype is primitive
- map whose subtype is primitive
- reference references another type
?

Reply:
No, from what I have seen these types can be complex (see the count schema for an example). However, our current schema resource doesn't handle this case well (see the count schema and the item count resource), so we don't need to handle it here.